### ISSUE TITLE
versions:set does not honour Maven's -f parameter

### ISSUE THREAD

**Issue by kalibyrn** [mojohaus/versions-maven-plugin#43]
If I have a pom called "everything-pom.xml" in the current directory (called "test") that looks like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.example.app</groupId>
    <artifactId>app-everything</artifactId>
    <packaging>pom</packaging>
    <version>4.1.0-SNAPSHOT</version>
	
</project>
```

and want to run the following command:

```
mvn -f everything-pom.xml versions:set -DnewVersion=5.0
```

I get an error saying the file "pom.xml" cannot be found.  It makes me think that "pom.xml" is hardcoded somewhere. It should probably be made aware of Maven's -f parameter.

```
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building app-everything 4.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- versions-maven-plugin:2.2:set (default-cli) @ app-everything ---
[INFO] Searching for local aggregator root...
[INFO] Local aggregation root: D:\test
[INFO] Processing change of com.example.app:app-everything:4.1.0-SNAPSHOT -> 5.0
[ERROR]
java.io.FileNotFoundException: D:\test\pom.xml (The system cannot find the file specified)
        at java.io.FileInputStream.open(Native Method)
        at java.io.FileInputStream.<init>(FileInputStream.java:146)
        at org.codehaus.plexus.util.xml.XmlReader.<init>(XmlReader.java:124)
        at org.codehaus.plexus.util.xml.XmlStreamReader.<init>(XmlStreamReader.java:67)
        at org.codehaus.plexus.util.ReaderFactory.newXmlReader(ReaderFactory.java:118)
        at org.codehaus.mojo.versions.api.PomHelper.readXmlFile(PomHelper.java:1613)
        at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.process(AbstractVersionsUpdaterMojo.java:319)
        at org.codehaus.mojo.versions.SetMojo.execute(SetMojo.java:251)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.953 s
[INFO] Finished at: 2015-12-11T14:17:36-05:00
[INFO] Final Memory: 12M/244M
[INFO] ------------------------------------------------------------------------
```

**Comment by prostagma** [comment]
Yes, ```"pom.xml"``` is really [hardcoded](https://github.com/mojohaus/versions-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/versions/SetMojo.java#L373). 
There already is the [#27](https://github.com/mojohaus/versions-maven-plugin/pull/27) pull request to fix this.

**Comment by u3r** [comment]
same problem here!
It doesn't ignore `-f` for all operations but the exception is annoying.

**Comment by khmarbaise** [comment]
Fixed with  PR #27

**Comment by tobias-hammerschmidt** [comment]
Any estimate on when 2.3 will be published? Also suffering from this issue with 2.2.
